tests/spark_autologging/ml/test_pyspark_ml_autologging.py

```diff
@@ -582,3 +598,163 @@ def test_gen_estimator_metadata(spark_session):  # pylint: disable=unused-argume
+    accruacy_original = mce.evaluate(pred_result, params={mce.metricName: "accuracy"})
+    assert np.isclose(accuracy, accruacy_original)
+    areaUnderROC_original = bce.evaluate(pred_result)
+    assert np.isclose(areaUnderROC, areaUnderROC_original)
+
+
+def test_multi_model_interleaved_fit_and_post_train_metric_call(dataset_iris_binomial):
+    mlflow.pyspark.ml.autolog()
+
+    estimator1 = LogisticRegression(maxIter=1, family="binomial", regParam=5.0, fitIntercept=False)
+    estimator2 = LogisticRegression(maxIter=5, family="binomial", regParam=5.0, fitIntercept=False)
+    eval_dataset1 = dataset_iris_binomial.sample(fraction=0.3, seed=1)
+    eval_dataset2 = dataset_iris_binomial.sample(fraction=0.3, seed=2)
+    mce = MulticlassClassificationEvaluator(metricName="logLoss")
+
+    with mlflow.start_run() as run1:
+        model1 = estimator1.fit(dataset_iris_binomial)
+
+    with mlflow.start_run() as run2:
+        model2 = estimator2.fit(dataset_iris_binomial)
+
+    pred1_result = model1.transform(eval_dataset1)
+    pred2_result = model2.transform(eval_dataset2)
+
+    logloss1 = mce.evaluate(pred1_result)
+    logloss2 = mce.evaluate(pred2_result)
+
+    metrics1 = get_run_data(run1.info.run_id).metrics
+    assert np.isclose(logloss1, metrics1["logLoss_eval_dataset1"])
+
+    metrics2 = get_run_data(run2.info.run_id).metrics
+    assert np.isclose(logloss2, metrics2["logLoss_eval_dataset2"])
+
+
+def test_meta_estimator_disable_post_training_autologging(dataset_regression):
+    mlflow.pyspark.ml.autolog()
+    lr = LinearRegression(solver="l-bfgs", regParam=0.01)
+    eval_dataset = dataset_regression.sample(fraction=0.3, seed=1)
+    lrParamMaps = [
+        {lr.maxIter: 1, lr.standardization: False},
+        {lr.maxIter: 200, lr.standardization: True},
+        {lr.maxIter: 2, lr.standardization: False},
+    ]
+    eva = RegressionEvaluator(metricName="rmse")
+    estimator = TrainValidationSplit(estimator=lr, estimatorParamMaps=lrParamMaps, evaluator=eva)
+
+    with mock.patch(
+        "mlflow.pyspark.ml._AutologgingMetricsManager.register_model"
+    ) as mock_register_model, mock.patch(
+        "mlflow.sklearn._AutologgingMetricsManager.is_metric_value_loggable"
+    ) as mock_is_metric_value_loggable, mock.patch(
+        "mlflow.pyspark.ml._AutologgingMetricsManager.log_post_training_metric"
+    ) as mock_log_post_training_metric, mock.patch(
+        "mlflow.pyspark.ml._AutologgingMetricsManager.register_prediction_input_dataset"
+    ) as mock_register_prediction_input_dataset:
+        with mlflow.start_run():
+            model = estimator.fit(dataset_regression)
+
+        model.transform(eval_dataset)
+
+        mock_register_model.assert_called_once()
+        mock_is_metric_value_loggable.assert_not_called()
+        mock_register_prediction_input_dataset.assert_not_called()
+        mock_log_post_training_metric.assert_not_called()
+
+
+def test_is_metrics_value_loggable():
```

Comment:
This test is similar with the one in sklearn autolog test.

After the follow-up AutologgingMetricManager refactor, we should remove this duplicated test.